tests/unit/test_operations/sql/test_required_variables.py

```
@@ -0,0 +1,129 @@
+        data_service,
+        table_name="AE",
+        column_data={
+            "STUDYID": ["TEST_STUDY", "TEST_STUDY", "TEST_STUDY"],
+            "AETERM": ["test", "test", "test"],
+        },
+        standards_context=standards_context,
+    )
+
+    params = SqlOperationParams(domain="AE", target=None, standards_context=standards_context)
+
+    operation = SqlOperationsFactory.get_service("required_variables", params, data_service)
+
+    # Mock the metadata retrieval method on the operation instance
+    with patch.object(
+        operation,
+        "_get_variables_metadata_from_standard_model",
+        return_value=mock_variables,
+    ):
+        result = operation.execute()
+        assert_operation_collection(operation, result, expected)
+
+
+def test_required_variables_exception_handling():
+    """Test get_model_filtered_variables when metadata retrieval fails"""
+    data_service = PostgresQLDataService.instance()
+    standards_context = SdtmStandardsContext(LibraryMetadataContainer())
+
+    # Add test dataset matching original test structure
+    PostgresQLDataService.add_test_dataset(
+        data_service,
+        table_name="AE",
+        column_data={
+            "STUDYID": ["TEST_STUDY", "TEST_STUDY", "TEST_STUDY"],
+            "AETERM": ["test", "test", "test"],
+        },
+        standards_context=standards_context,
+    )
+
+    params = SqlOperationParams(domain="AE", target=None, standards_context=standards_context)
+
+    operation = SqlOperationsFactory.get_service("required_variables", params, data_service)
+
+    # Mock the metadata method to raise an exception
+    with patch.object(
+        operation,
+        "_get_variables_metadata_from_standard_model",
+        side_effect=Exception("Metadata retrieval failed"),
+    ):
+        result = operation.execute()
+
+        # Should return empty collection on exception
+        assert_operation_collection(operation, result, [])
```

I'd argue it should just propagate the error, if the metadata isn't available the whole rule can't be executed

---

tweaked required_variables to throw error on metadata retrieval failure (and test_required_variables to check this)